runtime/src/utils/buffer.rs

```diff
@@ -111,11 +114,12 @@ impl<B: Blob> Buffer<B> {
         let bytes_to_read = std::cmp::min(self.buffer_size as u64, blob_remaining) as usize;
 
         // Read the data - we only need a single read operation since we know exactly how much data is available
-        self.blob
-            .read_at(&mut self.buffer[..bytes_to_read], self.blob_position)
+        let buffer = std::mem::take(&mut self.buffer);
+        self.buffer = self
```

Comment:
This is a little bit sus because on the last `refill`, `self.buffer` will have length < `self.buffer_size`. Will try to find a way to improve this.

Reply:
The code is correct as it is, but if it's modified, then this could be a foot gun. I added a comment flagging this for maintainers.